docs_src/threat-models/secret-store/threat_model.md

```diff
@@ -0,0 +1,353 @@
+
+#### General protections
+
+* The `root` user in a container is subject to namespace constraints and restricted set of [capabilities](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities).
+
+#### File system protections
+
+* Containers by default has no visibility to the host's file system and run with their own root file system that is supplied with the container. The container's file system can be augmented with docker volumes and bind mounts to the host file system to allow specific data sharing scenarios.
+* Containers can be started with tmpfs volumes that are local to that container instance. By default, all files in a container are remapped to an overlay file system stored as files under `/var/lib/docker` where they are observable on the host and stored persistently.
+* The root file system of a container can be mounted read-only.  For writable root file systems, each container gets a fresh copy of the root file system.
+* Content that must be persisted across container restarts must be stored in Docker volumes.
+* Docker volumes can be shared across multiple containers; however, the default "local" driver can only do such sharing when the containers are co-located on the same host.
+
+#### Interprocess communication protections
+
+* Docker containers do not share the host's network interface by default and instead is based on virtual ethernet adapters and bridges.  Network connectivity is strictly controlled via the docker-compose definition.
+* There are networking differences when running Docker on Windows or MacOS machines, due to the use of a hidden Linux virtual machine to actually run Docker.
+* There are few if any IPC restrictions between processes running in the same container due to lack of mandatory access controls.  Each service must run in its own container to ensure maximum service isolation.
+
+### Snap-based runtimes
+
+All services running within a single snap are assumed to be within the same trust boundary.  However, even in a snap, due to the use of mandatory access control, there are stronger-than-normal process isolation policies in place, as documented below.
+
+#### General protections
+
+* The `root` user in a snap is subject to namespace constraints and MAC rules enforced by Linux Security Modules (LSMs) configured as part of the snap.
+
+#### File system protections
+
+* Snaps run inside their own mount namespace, which is a [confined](https://github.com/snapcore/snapd/wiki/Snap-Execution-Environment) view of the host's file system where access to most paths is restricted. This includes sysfs and procfs.  Note: File system paths inside of the snap are homomorphic with the host's view of the file system - any files written in the snap are visible on the host.
+* All of the files in the snap are read-only with the exception if the below noted paths.  The contents of the snap itself are mounted read-only from a squashfs file system.
```

Comment:
if the below noted paths -> to the below noted paths??

Reply:
"of"